packages/0018-package-format.md

```diff
@@ -0,0 +1,163 @@
+
+Request for Contribution (RFC) 22 : _Package Format_
+
+## Code of conduct
+
+Please read and respect the [RFC Code of Conduct](https://github.com/umbraco/rfcs/blob/master/CODE_OF_CONDUCT.md).
+
+## Intended Audience
+
+The intended audience for this RFC is: Package developers and consumers.
+
+## Summary
+
+Umbraco currently has two package formats: 
+- The default Umbraco ZIP package format that can be generated from the backoffice
+- The standard .NET NuGet package format
+
+The purpose of this RFC is to suggest a way to bridge the gap and unify the two package formats while also looking closer at what a modern package format *should* offer.
+
+## Motivation
+
+Many package developers end up publishing their packages as both a Zip and NuGet packages since both formats have unique advantages. However, as a developer, working with NuGet packages are highly preferred as it has some essential features such as having automatic dependency handling and being fully integrated with application development pipelines such as Visual Studio and CI/CD platforms.
+
+We want to make it easier and simpler to create packages by only maintaining one format while getting the best features of both.
+
+## Detailed Design
+
+The high-level overview is that we want to remove the zip package format and move to a NuGet zip (.nupkg), and then make sure that that the benefits of the zip format is maintained:
+
+- Listing in the backoffice
+- Installing from the backoffice
+- Ability to install Umbraco content and schema
+- Ability to run code within the Umbraco context on install and uninstall
+- Creating a package from the backoffice
+
+We have a more detailed list of features we want the new package format to support below. If you feel something is missing please check the **Out of scope** section before commenting, we may have the same idea but want to leave it out for this first iteration!
+
+### Creating a package
+
+#### Umbraco content & media
+
+We want to ensure that we don't take a step back with functionality compared to what you can do today. One of the things you can do in an Umbraco Zip package but can't do in a NuGet package today is to include content and schema (and media with a lot of extra work). 
+
+We want to ensure that when you create a package in the backoffice you can include content and media (see schema below), to do that we will need to support serializing and deserializing content and media in the CMS.
+
+#### Umbraco schema
+
+Similar to the Umbraco content and media above, we want to keep the option of installing schema (document types, data types, templates, languages, etc.). 
+
+We also want to simplify the package creation approach when including Umbraco schema and content to **only allow you to pick all items**. Most packages are made on a specific site to develop that one package, we believe it will make it faster and simpler to create and manage packages. For packages based on files the option to pick files will stay as-is.
```

Comment:
I don't like this limitation. A package test website will often contain random test pages if the package is in any way related to the front-end so now we need another website that only has the package content types in there? Please consider keeping pickers for content types / data types / content.

Reply:
I will be the voice of the devil here. I would say package generator from backoffice should be removed at all, as, in my opinion, it should be just exporter which will export that to XML/JSON/Any format and package shouldn't be build from Backoffice. Packages should have the option to use Importer, but Content Importer should be a separate feature, not related to packages.

Reply:
That's a good point @PerplexDaniel!

It would still be possible to remove pages from the package xml so it would come down to what would benefit the most people.. 

As with everything else in this RFC nothing is set in stone yet, appreciate the perspectives!